Compiler/src/Compiler.cpp

```diff
@@ -1477,6 +1480,70 @@ struct Compiler
         }
     }
 
+    void compileExprInterpString(AstExprInterpString* expr, uint8_t target, bool targetTemp)
+    {
+        std::string formatString;
+
+        size_t stringsLeft = expr->strings.size;
+
+        for (AstArray<char> const& string : expr->strings)
+        {
+            std::string stringEscaped(string.data);
+
+            for (size_t characterIndex = 0; characterIndex < stringEscaped.size(); ++characterIndex)
+            {
+                if (stringEscaped[characterIndex] == '%')
+                {
+                    stringEscaped.insert(characterIndex, 1, '%');
+                    characterIndex++;
+                }
+            }
+
+            formatString += stringEscaped;
+
+            stringsLeft--;
+
+            if (stringsLeft > 0)
+                formatString += "%*";
+        }
+
+        auto formatStringSize = formatString.size();
+
+        // We can't use formatStringRef.data() directly, because short strings don't have their data
+        // pinned in memory, so when interpFormatStrings grows, these pointers will move and become invalid.
+        std::shared_ptr<char[]> formatStringPtr(new char[formatStringSize]);
```

Comment:
What was the problem with unique_ptr? You might need to use `std::move` when adding it to the vector, but it should work otherwise.

Reply:
No matter what I do, `emplace_back` with a unique_ptr always reports as `= delete`.

I can use `push_back` with `std::move`, but then when I create `AstArray<char> formatStringArray{formatStringPtr.get(), formatStringSize};`, I'm using the value after move, which feels like something I should not be doing.

If I then add `auto formatStringRef = interpFormatStrings.back();`, then I get the `unique_ptr(const unique_ptr&) = delete;` error again.

Reply:
Interesting. Let's table this for now, this should be possible to fix but it's easier to do it later once everything else is resolved.

Reply:
I've updated the code to use std::unique_ptr.